### PR TITLE
Bring back game log parser

### LINKED ISSUE #408: Re-integrate game-log-parser Scala/SBT project
As there's no benefit to having it as a separate library.

- Git/SBT project: https://github.com/ActionFPS/game-log-parser
  - Take GLP repo and move its root contents into a sub-directory `game-log-parser`.
  - Merge new GLP repo into ActionFPS repo (this repo) using this technique https://stackoverflow.com/a/10548919 (but put into game-log-parser directory).
  - Integrate GLP's `build.sbt` into ActionFPS `build.sbt`, excluding all the publishing features. No `built.sbt` should remain afterwards. Put dependencies into `Dependencies.scala`.
  - Remove reference to game-log-parser in `Dependencies.scala` of the SBT build, instead referring to the local project
  - I may have missed something else that needs to be removed.
- Remove references from Docs: https://github.com/ActionFPS/ActionFPS-Book/blob/master/portal-development-guide.md

I'll pay $40 for this ([ScalaWilliam Work](http://work.scalawilliam.com/integrate-project-back/)).
Rules: http://work.scalawilliam.com/rules/
Please post here "**I'm taking on this**" to lock the task to yourself.

### PR DESCRIPTION
**Pull request by omainegra**
What's done so far
  - [x] Merge new GLP repo into ActionFPS (using `git subtree`)
  - [x] Integrate GLP's `build.sbt` into ActionFPS `build.sbt`
  - [x] Remove reference to game-log-parser in `Dependencies.scala` of the SBT build, instead referring to the local project

This closes #408

### CODE REVIEW

File: game-log-parser/project/plugins.sbt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+resolvers += "Typesafe repository" at "http://repo.typesafe.com/typesafe/releases/"
```
Comment (ScalaWilliam) [inline game-log-parser/project/plugins.sbt]: No longer needed.

File: game-log-parser/project/build.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+sbt.version=0.13.13
```
Comment (ScalaWilliam) [inline game-log-parser/project/build.properties]: No longer needed

File: game-log-parser/.gitignore [hunk 3]
```diff
@@ -0,0 +1,5 @@
+*.tmp
```
Comment (ScalaWilliam) [inline game-log-parser/.gitignore]: Can be merged into root .gitignore I believe

**Comment by ScalaWilliam** [comment]
- Root project needs to aggregate the new two subprojects (`game-parser/testOnly *UserHostParserSpec*` runs the test but `testOnly *UserHostParserSpec*` doesn't)
- Can you make reference to the issue from the pull request description? GitHub will link them automatically

File: game-log-parser/README.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+The less context per each line, the better. For example, instead of showing
+`[1.2.3.4] w00p|Drakas fragged w00p|pwns`, let's remove the indirection by
+including an IP address of the other person, as well as a CN of that person.
+
+Including the current team? Not sure if that's useful.
+
+Let's just start from the most basic stuff, which is time.
+
+# Release procedure
```
Comment (ScalaWilliam) [inline game-log-parser/README.md]: This section no longer relevant

File: game-log-parser/README.md [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+as IPs may change due to migrations etc.
+
+So perhaps this is the common format we should look at producing from now on, in all cases.
+
+Currently we don't support the 'Ideal format' for any of the tools.
+
+
+## Tools
+
+### <a href="blob/master/iso_local_datetime.pl">iso_local_datetime.pl</a>
```
Comment (ScalaWilliam) [inline game-log-parser/README.md]: These links will not work any longer - I just say remove them as they're a maintenance nightmare.

File: game-log-parser/app/src/main/scala/GameLogParserApp.scala [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import com.actionfps.gameparser.GameScanner
```
Comment (ScalaWilliam) [inline game-log-parser/app/src/main/scala/GameLogParserApp.scala]: The app disappeared from the build :-)

File: game-log-parser/README.md (hunk 4 above)
Comment (omainegra) [reply game-log-parser/README.md]: Done

File: game-log-parser/app/src/main/scala/GameLogParserApp.scala (hunk 6 above)
Comment (omainegra) [reply game-log-parser/app/src/main/scala/GameLogParserApp.scala]: Brought back: `game-log-parser-app/run`

File: game-log-parser/.gitignore (hunk 3 above)
Comment (omainegra) [reply game-log-parser/.gitignore]: Done

File: game-log-parser/project/plugins.sbt (hunk 1 above)
Comment (omainegra) [reply game-log-parser/project/plugins.sbt]: Removed `project` folder

File: game-log-parser/project/build.properties (hunk 2 above)
Comment (omainegra) [reply game-log-parser/project/build.properties]: Removed `project` folder

File: game-log-parser/README.md (hunk 5 above)
Comment (omainegra) [reply game-log-parser/README.md]: Done

**Comment by ScalaWilliam** [comment]
@omainegra please see comments -> https://github.com/ScalaWilliam/ActionFPS/issues/408#issuecomment-338061546

**Pull request merged**